cli/loader_lib.c

```diff
@@ -153,6 +154,82 @@ JL_DLLEXPORT const char * jl_get_libdir()
     return lib_dir;
 }
 
+#ifdef LIBSTDCXXPROBE
+// Write the path to the libstdcxx to load to "path", which should be a buffer of at least
+// JL_PATH_MAX, and the length of that string (excluding null terminator) to "written"
+// Returns 1 on success or 0 on error, in which case the caller should abort the process.
+#define JL_DEFAULT_STDCXXLIB "libstdc++.so.6"
```

Comment:
We need to calculate the path to the bundled `libstdc++.so` at runtime.  Let's make a function called `get_bundled_libstdcxx()` or something like that, and have it return the equivalent of `joinpath(jl_get_libdir(), "julia", "bundled", "libstdc++.so")`.  We can then alter the `libstdc++.so` installation rules to install to `lib/julia/bundled` instead of just `lib/`.

## `libstdc++` installation rules

There are two stages of `libstdc++` installation: there's where it lives after you've run `make` (and Julia lives in the build tree, in `usr/bin/julia`), and where it lives after `make install` (which is the portable binary installation).  I for one absolutely hate that there is a difference, I would much rather that `make` created a build tree that is identical to what gets shipped when you do `make install`, but, unfortunately, it's a big lift to properly unify things, and there are some good reasons (such as multiarch libdir layouts) for why it exists, so c'est la vie, I suppose).

### `make` time:

When you run `make`, we'll run the steps in [`deps/csl.mk`](https://github.com/JuliaLang/julia/blob/68632490a2ac52320734acbbbea6e3abc8bb2b3f/deps/csl.mk) (which stands for Compiler Support Libraries), which will either [download a binary package](https://github.com/JuliaLang/julia/blob/68632490a2ac52320734acbbbea6e3abc8bb2b3f/deps/csl.mk#L107) of standard libraries (snarfed from [CompilerSupportLibraries_jll](https://github.com/JuliaBinaryWrappers/CompilerSupportLibraries_jll.jl)), or [steal the binaries from your current compilers](https://github.com/JuliaLang/julia/blob/68632490a2ac52320734acbbbea6e3abc8bb2b3f/deps/csl.mk#L66), if you've opted to build everything locally.

To adjust this step, you need to redirect the `libstdc++` installation location.  For the from-source installation this is fairly simple; you need to replace [this invocation](https://github.com/JuliaLang/julia/blob/68632490a2ac52320734acbbbea6e3abc8bb2b3f/deps/csl.mk#L66) with one that installs to that "bundled" directory (something like `$(build_private_shlibdir)/bundled`) instead of the main shared library directory.

For the bb install step, it's a little trickier, we want to basically move the `libstdc++` library after it's been unpacked from the downloaded tarball.  We can make a new makefile target called `$(build_private_shlibdir)/bundled/$(call versioned_libname,libstdc++,6)` that `install-csl` will depend on, and have that makefile target copy the library from where it usually sits to where it should actually sit.  And, of course, we only want to do this on glibc platforms.  ;)

### `make install` time

Since you've gone through the work of getting things set up correctly at `make` time, the `make install` time should be quite easy.  Remove `libstdc++` from [this list](https://github.com/JuliaLang/julia/blob/68632490a2ac52320734acbbbea6e3abc8bb2b3f/Makefile#L213) which will try to copy it along with other libraries, and instead create a custom copy command [here](https://github.com/JuliaLang/julia/blob/68632490a2ac52320734acbbbea6e3abc8bb2b3f/Makefile#L290) that copies the entire `bundled` directory over to `$(DESTDIR)$(private_libdir)`.

Currently, we [add `libstdc++` onto our `JL_PRIVATE_LIBS` list](https://github.com/JuliaLang/julia/blob/68632490a2ac52320734acbbbea6e3abc8bb2b3f/Makefile#L213), but we want to avoid that, because [those all get installed to `lib`](https://github.com/JuliaLang/julia/blob/68632490a2ac52320734acbbbea6e3abc8bb2b3f/Makefile#L291-L297) and [then `lib/julia`]()